samples/semantickernel-concepts/semantickernel-syntax-examples/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/memory/AzureAISearchVectorStore.java

```diff
@@ -136,18 +136,26 @@ public static void dataStorageWithAzureAISearch(
             .then(storeData(collection, embeddingGeneration, sampleData()))
             .block();
 
-        // Query the Azure AI Search client for results
-        // This might take a few seconds to return the best result
-        var result = searchClient.getSearchAsyncClient(collectionName)
-            .search("How to get started with the Semantic Kernel?")
-            .blockFirst();
+        // Search for results
+        // Might need to wait for the data to be indexed
+        var results = search("How to get started", collection, embeddingGeneration).block();
+        var searchResult = results.get(0);
+        System.out.printf("Search result with score: %f.%n Link: %s, Description: %s%n",
+                searchResult.getScore(), searchResult.getRecord().link, searchResult.getRecord().description);
+    }
+
+
+    private static Mono<List<VectorSearchResult<GitHubFile>>> search(
+            String searchText,
+            AzureAISearchVectorStoreRecordCollection<GitHubFile> recordCollection,
+            OpenAITextEmbeddingGenerationService embeddingGeneration) {
 
-        GitHubFile gitHubFile = result.getDocument(GitHubFile.class);
-        System.out.println("Best result: " + gitHubFile.description + ". Link: " + gitHubFile.link);
+        return embeddingGeneration.generateEmbeddingsAsync(Collections.singletonList(searchText))
+                .flatMap(r -> recordCollection.searchAsync(r.get(0).getVector(), null));
```

Comment:
what does the get(0) refer to here? Just wondering why only first elemen

Reply:
Our embedding generation service receives a list of strings to generate the embeddings for. Only one was sent, that's the one retrieved.